rq/dependency.py

```diff
@@ -0,0 +1,26 @@
+from .job import Job
+from redis.client import Pipeline
+from typing import List
+from redis.exceptions import WatchError
+
+
+class Dependency:
+    @classmethod
+    def get_jobs_with_dependencies_met(cls, jobs: List['Job'], pipeline: Pipeline):
+        pipe = pipeline if pipeline else cls.connection
```

Comment:
`pipeline` is required so no need to fallback to `cls.connection`

Reply:
Removed